akka-docs/src/main/paradox/testing-typed.md

```diff
@@ -29,4 +29,163 @@ Maven
+
+Anonymously:
+
+Scala
+:  @@snip [BasicSyncTestingSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/testing/sync/BasicSyncTestingSpec.scala) { #test-anonymous-child }
+
+Java
+:  @@snip [BasicSyncTestingTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java) { #test-anonymous-child } 
+
+### Sending messages
+
+For testing sending a message a `TestInbox` is created that provides an `ActorRef` and methods to assert against the
+messages that have been sent to it.
+
+Scala
+:  @@snip [BasicSyncTestingSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/testing/sync/BasicSyncTestingSpec.scala) { #test-message }
+
+Java
+:  @@snip [BasicSyncTestingTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java) { #test-message } 
+
+Another use case is sending a message to a child actor you can do this by looking up the 'TestInbox' for
+a child actor from the 'BehaviorTestKit':
+
+Scala
+:  @@snip [BasicSyncTestingSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/testing/sync/BasicSyncTestingSpec.scala) { #test-child-message }
+
+Java
+:  @@snip [BasicSyncTestingTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java) { #test-child-message } 
+
+For anonymous children the actor names are generated in a deterministic way:
+
+Scala
+:  @@snip [BasicSyncTestingSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/testing/sync/BasicSyncTestingSpec.scala) { #test-child-message-anonymous }
+
+Java
+:  @@snip [BasicSyncTestingTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/testing/sync/BasicSyncTestingTest.java) { #test-child-message-anonymous } 
+
+### Testing other effects
+
+The `BehaviorTestkit` keeps track other effects you can verify, look at the sub-classes of `akka.typed.testkit.Effect`
+ 
+ * SpawnedAdapter
+ * Stopped
+ * Watched
+ * Unwatched
+ * Scheduled
+ 
+See the other public methods and API documentation on `BehaviourTestkit` for other types of verification.
+   
+## Asynchronous testing
+
+Asynchronous testing uses a real `ActorSystem` that allows you to test your Actors in a realistic environment. 
+
+The minimal setup consists of the test procedure, which provides the desired stimuli, the actor under test, 
+and an actor receiving replies. Bigger systems replace the actor under test with a network of actors, apply stimuli 
+at varying injection points and arrange results to be sent from different emission points, but the basic principle stays 
+the same in that a single procedure drives the test.
+
+@scala[
+If you are familiar with testing untyped actors there are some differences:
+
+* No `ImplicitSender`, instead create `TestProbe[T]` that will be in the messages sent to your actor
+* No test actor so all assertions are done with `TestProbe`s
+]
```

Comment:
You have to do the div trick to make a section for scala only, see:

```
@@@ div { .group-scala }
@@@
```

like in https://github.com/akka/akka-http/pull/1700#issuecomment-353601706

Reply:
Thanks